src/index.ts

```diff
@@ -1,5 +1,50 @@
-"use strict";
+import { SessionMiddlewareFactory } from "./SessionMiddlewareFactory";
+import { SessionHandlerConfig } from "./SessionHandlerConfig";
+import { SessionStore } from "./session/SessionStore";
+import { Cache } from "./cache/Cache";
+import { RequestHandler } from "express";
 
-import SessionMiddlewareFactory from "./SessionMiddlewareFactory";
 
-export = SessionMiddlewareFactory;
+export interface CHSessionService {
+    readonly config: SessionHandlerConfig;
+    readonly sessionStore: SessionStore;
+    readonly sessionHandler: RequestHandler;
+}
+
+let handler: CHSessionService;
+
+class CHSessionServiceImpl implements CHSessionService {
+
+    private static impl: CHSessionServiceImpl;
+    private sessionMiddlewareFactory: SessionMiddlewareFactory;
+
+    public readonly sessionStore: SessionStore;
+    public readonly sessionHandler: RequestHandler;
+
+    private constructor(public readonly config: SessionHandlerConfig) {
+        const redis = Cache.redisInstance(config);
```

Comment:
This is super confusing. Cache class is used in two ways:
1) to produce redis instance
2) to wrap redis instance

This hides real purpose of the class. Is it more focused on hiding implementation details (raw redis client) or is it not?